tools/driver/p4c_src/main.py

```diff
@@ -184,20 +192,30 @@ def main():
         print display_supported_targets(cfg)
         sys.exit(0)
 
+    # When using --help-* options, we don't necessarily need to pass an input file
+    # However, by default the driver checks the input and fails if it does not exist.
+    # Also, loading the backend configuration files requires a source file to be set,
+    # so in that case, we set the input to dummy.p4 and expect that the backend itself
+    # will do its printing and ignore the input file.
+    # If not, the compiler will error out anyway.
+    checkInput = True if not opts.help_pragmas else False
+
     input_specified = False
     if opts.source_file:
         input_specified = True
     if (os.environ['P4C_BUILD_TYPE'] == "DEVELOPER"):
         if opts.json_source:
             input_specified = True
-    if not input_specified:
+    if checkInput and not input_specified:
         parser.error('No input specified.')
+    elif not checkInput:
+        opts.source_file = "dummy.p4"
```

Comment:
This risks having confusing/misleading error messages. I think a better approach would be to thread `checkInput` through to the relevant places. Granted, that would be a more disruptive change.

Reply:
The changes are quite significant to handle at this level vs. local in each backend. I took the lazy approach and as backends implement this change, the confusing error message disappears.